src/backend/src/services/projects.services.ts

```diff
@@ -506,4 +507,59 @@ export default class ProjectsService {
 
     return projectTransformer(deletedProject);
   }
+
+  /**
+   * Toggles a user's favorite status on a projects
+   * @param projectId the project id to be favorited/unfavorited
+   * @param user the user who is changing the role
+   * @param favorite boolean representing favoriting/unfavoriting
+   * @returns the project that the user has favorited/unfavorited
+   * @throws if the project id doesn't exist
+   */
+  static async toggleFavorite(wbsNumber: WbsNumber, user: PrismaUser): Promise<Project> {
+    if (!isProject(wbsNumber)) throw new HttpException(400, `${wbsPipe(wbsNumber)} is not a valid project WBS #!`);
+    const { carNumber, projectNumber, workPackageNumber } = wbsNumber;
+
+    const projectInfo = await prisma.project.findFirst({
+      where: {
+        wbsElement: {
+          carNumber,
+          projectNumber,
+          workPackageNumber
+        }
+      },
+      include: {
+        ...projectQueryArgs.include,
+        favoritedBy: true
+      }
+    });
+
+    if (!projectInfo) throw new NotFoundException('WBS Element', wbsPipe(wbsNumber));
+    const favorited = projectInfo.favoritedBy.filter((allUsers) => allUsers.userId === user.userId).length;
```

Comment:
this is confusing id map them to be user ids then do contains or something like that

Reply:
ok